### PR TITLE
Bump Ember CLI to 2.14.0

### PR DESCRIPTION
**Pull request by jherdman**
In trying to track down a bug, I noticed that I could no longer build the "ember-beta" scenario. I've upgraded Ember CLI as part of resolving this problem. I pretty much blindly followed the upgrade instructions, please let me know if I can tweak this at all to suit your tastes and/or needs.

Relatedly, this will fail with ember-beta! This is good for me (see above), but not so good for this pull request.

### CODE REVIEW

**Comment by jherdman** [comment]
One more question. Since NPM5 creates a package-lock.json file, would you like this in the PR as well?

**Comment by jherdman** [comment]
Example fail in ember-beta:

```
not ok 12 PhantomJS 2.1 - Integration | Component | power-datepicker: When passed `closeOnSelect=false` selecting a date doesn't close the datepicker
    ---
        actual: >
            false
        expected: >
            true
        stack: >
            exception@http://localhost:7357/assets/test-support.js:9167:46
            adapterDispatch@http://localhost:7357/assets/vendor.js:69552:22
            dispatchError@http://localhost:7357/assets/vendor.js:51563:23
            invokeWithOnError@http://localhost:7357/assets/vendor.js:34306:24
            flush@http://localhost:7357/assets/vendor.js:34174:27
            flush@http://localhost:7357/assets/vendor.js:34365:36
            end@http://localhost:7357/assets/vendor.js:34473:47
            run@http://localhost:7357/assets/vendor.js:34551:29
            run$1@http://localhost:7357/assets/vendor.js:51924:32
            clickEventSequence@http://localhost:7357/assets/test-support.js:8233:8
            click@http://localhost:7357/assets/test-support.js:8261:23
            clickTrigger@http://localhost:7357/assets/tests.js:220:38
            http://localhost:7357/assets/tests.js:507:42
            runTest@http://localhost:7357/assets/test-support.js:4028:34
            run@http://localhost:7357/assets/test-support.js:4014:13
            http://localhost:7357/assets/test-support.js:4220:15
            advance@http://localhost:7357/assets/test-support.js:3673:26
            begin@http://localhost:7357/assets/test-support.js:5485:27
            http://localhost:7357/assets/test-support.js:4599:11
        message: >
            TypeError: undefined is not a constructor (evaluating 'context.value()')
```

This is congruent with what I'm seeing in my application.

File: .travis.yml [hunk 1]
```diff
@@ -1,17 +1,16 @@
...
 
 cache:
-  directories:
-    - $HOME/.npm
-    - $HOME/.cache # includes bowers cache
+  yarn: true
 
 env:
   # we recommend testing LTS's and latest stable release (bonus points to beta/canary)
+  - EMBER_TRY_SCENARIO=ember-lts-2.4
```
Comment (cibernox) [inline .travis.yml]: Please, leave it 2.8+

File: config/ember-try.js [hunk 2]
```diff
@@ -2,6 +2,22 @@
 module.exports = {
   scenarios: [
     {
+      name: 'ember-lts-2.4',
```
Comment (cibernox) [inline config/ember-try.js]: 2.8+

**Comment by cibernox** [comment]
The error in beta is actually due to a bug in ember: https://github.com/emberjs/ember.js/issues/15470

I added a couple minor changes. If tests are green in stable, don't worry about beta, it's not a problem in the addon.

File: .travis.yml [hunk 3]
```diff
@@ -1,14 +1,12 @@
 ---
 language: node_js
 node_js:
-  - "6"
+  - "4"
```
Comment (jherdman) [inline .travis.yml]: I'll revert this change too.

**Comment by jherdman** [comment]
I've made the changes as requested. Thank you for the info regarding the bug!

**Comment by cibernox** [comment]
Thanks

**Pull request merged**